src/compiler/checker.ts

```diff
@@ -4333,6 +4338,10 @@ module ts {
             return !!getPropertyOfType(type, "0");
         }
 
+        function isTupleType(type: Type) : boolean {

```

Comment:
doesn't always have `elementTypes`, which makes it hard to determine its tuple "length"